crates/polars-time/src/chunkedarray/rolling_window/dispatch.rs

```diff
@@ -267,34 +273,52 @@ pub trait SeriesOpsTime: AsSeries {
         by: &Series,
         options: RollingOptionsDynamicWindow,
     ) -> PolarsResult<Series> {
-        let s = self.as_series().clone();
+        let original_type = self.as_series().dtype();
+        let mut s = self.as_series().clone();
+        if s.dtype().is_bool() {
```

Comment:
I really don't think we should cast here, but rather have a specialize algorithm for booleans.

Reply:
I've implemented a specialized `rolling_min` and `rolling_max` algorithm for booleans, but the unsorted `by` column in `rolling_min_by` and `rolling_max_by` seems a bit tricky to me.

Maybe I can work on that later in another PR?